packages/devtools_app/lib/src/shared/memory/retainers.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+
+typedef IsRetainer = bool Function(int index);
+typedef References = List<int> Function(int index);
+typedef ShallowSize = int Function(int index);
+
+typedef ShortestRetainersResult = ({
+  List<int> retainers,
+  List<int>? retainedSizes,
+});
+
+ShortestRetainersResult findShortestRetainers(
+  int graphSize,
+  int rootIndex,
+  IsRetainer isRetainer,
+  References refs,
+  ShallowSize shallowSize, {
```

Comment:
make these required and named so that call sites of this method are easier to read